docs/dns-client/configuration.md

```diff
@@ -0,0 +1,200 @@
+---
+title: Configuration file
+sidebar_position: 2
+---
+
+See file [`config.dist.yml`][dist] for a full example of a [YAML][yaml] configuration file with comments.
+
+<!--
+    TODO(a.garipov): Find ways to add IDs to individual list items.
+ -->
+
+[dist]: https://github.com/AdguardTeam/AdGuardDNSClient/blob/master/config.dist.yaml
+[yaml]: https://yaml.org/
+
+## `dns` {#dns}
+
+The `dns` object configures the behavior of DNS server. It has the following properties:
+
+### `cache` {#dns-cache}
+
+The `cache` object configures caching the results of querying DNS. It has the following properties:
+
+- `enabled`: Whether or not the DNS results should be cached.
+
+    **Example:** `true`
+
+- `size`: The maximum size of the DNS results cache as human-readable data size. It must be greater than zero if `enabled` is `true`.
+
+    **Example:** `128MB`
+
+- `client_size`: The maximum size of the DNS results cache for each configured client’s address or subnetwork as human-readable data size. It must be greater than zero if `enabled` is `true`.
+
+    **Example:** `4MB`
+
+### `server` {#dns-server}
+
+The `server` object configures the handling of incoming requests. It has the following properties:
+
+- `listen_addresses`: The set of addresses with ports to listen on.
+
+    **Property example:**
+
+    ```yaml
+    'listen_addresses':
+      - address: '127.0.0.1:53'
+      - address: '[::1]:53'
+    ```
+
+### `bootstrap` {#dns-bootstrap}
+
+The `bootstrap` object configures the resolving of [upstream](#dns-upstream) servers addresses. It has the following properties:
+
+- `servers`: The list of servers to use for resolving upstream servers hostnames.
+
+    **Property example:**
+
+    ```yaml
+    'servers':
+      - address: '8.8.8.8:53'
+      - address: '192.168.1.1:53'
+    ```
+
+- `timeout`: The timeout for bootstrap DNS requests as a human-readable duration.
+
+    **Example:** `2s`
+
+### `upstream` {#dns-upstream}
+
+The `upstream` object configures the actual resolving of requests. It has the following properties:
+
+- `groups`: The set of upstream servers keyed by the group’s name. It has the following fields:
+
+    - `address`: The upstream server’s address.
+
+        **Example:** `'8.8.8.8:53'`
+
+    - `match`: The list of criteria to match the request against. Each entry may contain the following properties:
+
+        - `question_domain`: The domain or a suffix of the domain that the set of upstream servers should be used to resolve.
```

Comment:
This section causes issues with `docusaurus` 2.4.3 on my machine.  I wasn't able to find a way to make both `docusaurus` and `markdownlint` happy about it.